Source/WebCore/platform/network/curl/OpenSSLHelper.cpp

```diff
@@ -301,19 +301,76 @@ static void getSubjectAltName(const X509* x509, Vector<String>& dnsNames, Vector
             if (value->d.iPAddress->length == 4)
                 ipAddresses.append(makeString(data[0], ".", data[1], ".", data[2], ".", data[3]));
             else if (value->d.iPAddress->length == 16) {
-                StringBuilder ipAddress;
+                String ipAddress[8];
                 for (int i = 0; i < 8; i++) {
-                    ipAddress.append(makeString(hex(data[0] << 8 | data[1], 4)));
-                    if (i != 7)
-                        ipAddress.append(":");
+                    ipAddress[i] = makeString(hex(data[0] << 8 | data[1], 4, Lowercase));
                     data += 2;
                 }
-                ipAddresses.append(ipAddress.toString());
+                String ipv6 = canonicalizeIPv6Address(ipAddress);
+                ipAddresses.append(ipv6);
             }
         }
     }
 }
 
+String canonicalizeIPv6Address(String *ipInput) 
```

Comment:
This signature is too open ended.  It could point to any number of Strings, and they could contain anything.  I think it would be much better to pass a const std::array<uint16_t, 8>&

Reply:
Addressed in https://github.com/WebKit/WebKit/pull/3791/commits/75284d26b864b8943726f00329e296f662dbcdf3